sdk/messaging/azservicebus/internal/rpc.go

```diff
@@ -147,45 +140,57 @@ func NewRPCLink(args RPCLinkArgs) (*rpcLink, error) {
 
 	link.sender = sender
 	link.receiver = receiver
-	link.messageAccept = receiver.AcceptMessage
 
 	return link, nil
 }
 
+const responseRouterShutdownMessage = "Response router has shut down"
+
 // startResponseRouter is responsible for taking any messages received on the 'response'
 // link and forwarding it to the proper channel. The channel is being select'd by the
 // original `RPC` call.
 func (l *rpcLink) startResponseRouter() {
+	defer azlog.Writef(l.logEvent, responseRouterShutdownMessage)
+
 	for {
 		res, err := l.receiver.Receive(context.Background())
 
-		// You'll see this when the link is shutting down (either
-		// service-initiated via 'detach' or a user-initiated shutdown)
-		if isClosedError(err) {
-			l.broadcastError(err)
-			break
+		if err != nil {
+			// if the link or connection has a malfunction that would require it to restart then
+			// we need to bail out, broadcasting to all affected callers/consumers.
+			if GetRecoveryKind(err) != RecoveryKindNone {
+				azlog.Writef(l.logEvent, "Error in RPCLink, stopping response router: %s", err.Error())
+				l.broadcastError(err)
+				break
+			}
+
+			azlog.Writef(l.logEvent, "Non-fatal error in RPCLink, starting to receive again: %s", err.Error())
+			continue
 		}
 
 		// I don't believe this should happen. The JS version of this same code
 		// ignores errors as well since responses should always be correlated
 		// to actual send requests. So this is just here for completeness.
 		if res == nil {
+			azlog.Writef(l.logEvent, "RPCLink received no error, but also got no response")
 			continue
 		}
 
 		autogenMessageId, ok := res.Properties.CorrelationID.(string)
 
 		if !ok {
-			// TODO: it'd be good to track these in some way. We don't have a good way to
-			// forward this on at this point.
+			azlog.Writef(l.logEvent, "RPCLink message received without a CorrelationID %v", res)
 			continue
 		}
 
 		ch := l.deleteChannelFromMap(autogenMessageId)
 
-		if ch != nil {
-			ch <- rpcResponse{message: res, err: err}
+		if ch == nil {
+			azlog.Writef(l.logEvent, "RPCLink had no response channel for correlation ID %v", autogenMessageId)
```

Comment:
It should not - this falls int he bucket of "this should never happen" since it implies that the service is sending us a response for a message we did not send.

If it does generate noise that's actually a good thing because something bad is happening. And since nobody is actually seeing this error (ie, there's no channel to report it to) logging is a decent solution.